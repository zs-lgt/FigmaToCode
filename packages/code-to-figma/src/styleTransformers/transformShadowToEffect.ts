import { colorToRGBA } from './transformColors';
import { ShadowProperties, BlendStyleProperties } from './transformBlendProperties';

export const transformShadowToEffect = (styles: Partial<BlendStyleProperties>): Effect[] => {
    let shadows: Partial<ShadowProperties>[] = [styles];
    if ('shadows' in styles && Array.isArray(styles.shadows)) {
        shadows = styles.shadows;
    }
    return shadows.map(shadow => ({
        type: shadow.shadowType != null ? shadow.shadowType : 'DROP_SHADOW',
        color: colorToRGBA(shadow.shadowColor, shadow.shadowOpacity),
        offset: shadow.shadowOffset
            ? {
                  x: shadow.shadowOffset.width,
                  y: shadow.shadowOffset.height
              }
            : {
                  x: 0,
                  y: 0
              },
        radius: shadow.shadowRadius || 0,
        spread: shadow.shadowSpread || 0,
        visible: true,
        blendMode: 'NORMAL'
    }));
};
