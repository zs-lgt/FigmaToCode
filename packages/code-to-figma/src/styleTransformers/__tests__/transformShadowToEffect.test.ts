import { transformShadowToEffect } from '../transformShadowToEffect';

describe('transformShadowToEffect', () => {
    it('opacity from rgba', () => {
        const result = transformShadowToEffect({
            shadowColor: 'rgba(52,208,88,0.4)'
        });
        expect(result).toMatchSnapshot();
    });
});
