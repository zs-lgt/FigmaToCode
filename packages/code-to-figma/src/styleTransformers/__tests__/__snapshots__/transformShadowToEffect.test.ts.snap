// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformShadowToEffect opacity from rgba 1`] = `
Array [
  Object {
    "blendMode": "NORMAL",
    "color": Object {
      "a": 0.4,
      "b": 0.34509803921568627,
      "g": 0.8156862745098039,
      "r": 0.20392156862745098,
    },
    "offset": Object {
      "x": 0,
      "y": 0,
    },
    "radius": 0,
    "spread": 0,
    "type": "DROP_SHADOW",
    "visible": true,
  },
]
`;
